vowpalwabbit/feature_group.cc

```diff
@@ -140,13 +248,50 @@ bool features::sort(uint64_t parse_mask)
   apply_permutation_in_place(values, dest_index_vec);
```

Comment:
Worried about allocations that can happen in steady state:
1) sort operation (seems avoidable)
2) flatten_namespace

Reply:
Sort is very rarely used, it is only used with `limit` and `sort_features`. It is also done only once per example, so in practice this will not matter

Reply:
As discussed the apply function was rolled into one to avoid reallocating the done vec